<?xml version="1.0" ?>

<launch>
  <arg name="load_gripper" default="true" />
  <arg name="robot_ip" />
  <arg name="publish_rate" default="30" />

  <param unless="$(arg load_gripper)" name="robot_description" command="$(find xacro)/xacro --inorder $(find franka_description)/robots/panda_arm.urdf.xacro" />
  <param if="$(arg load_gripper)" name="robot_description" command="$(find xacro)/xacro --inorder $(find franka_description)/robots/panda_arm_hand.urdf.xacro" />

  <node name="robot_joint_state_publisher" pkg="franka_visualization" type="robot_joint_state_publisher" output="screen">
    <rosparam command="load" file="$(find franka_visualization)/config/robot_settings.yaml" />
    <param name="robot_ip" value="$(arg robot_ip)" />
    <param name="publish_rate" value="$(arg publish_rate)" />
  </node>

  <node name="gripper_joint_state_publisher" pkg="franka_visualization" type="gripper_joint_state_publisher" output="screen" if="$(arg load_gripper)">
    <rosparam command="load" file="$(find franka_visualization)/config/gripper_settings.yaml" />
    <param name="robot_ip" value="$(arg robot_ip)" />
    <param name="publish_rate" value="$(arg publish_rate)" />
  </node>

  <node name="joint_state_publisher" type="joint_state_publisher" pkg="joint_state_publisher" output="screen">
    <param name="rate" value="$(arg publish_rate)" />
    <rosparam param="source_list">[robot_joint_state_publisher/joint_states, gripper_joint_state_publisher/joint_states]</rosparam>
  </node>

  <node name="robot_state_publisher" pkg="robot_state_publisher" type="state_publisher" />
  <node pkg="rviz" type="rviz" output="screen" name="rviz" args="-d $(find franka_visualization)/launch/franka_visualization.rviz"/>
</launch>
